inifile.cc

```diff
@@ -169,23 +161,18 @@ inifile_load_file(gbfile* fin, inifile_t* inifile, const char* myname)
 static char*
 inifile_find_value(const inifile_t* inifile, const char* sec_name, const char* key)
```

Comment:
Ideally, the meat of this becomes a entry[val].data()  lookup. Maybe it's a two-stage thing to deal with each section, but maybe the internal storage becomes "Section:Entry" so you glom section and name together and can do a single lookup, at the cost of not being able to have a colon (or whatever) in the section or key names. But I'd bed this whole function could be a few lines long and exhibit way better than O(sections*N) lookups and stores. (Duly noting that readability and not performance is my actual point...)